### PR TITLE
Update to PBDB API version 1.2 and replace `raster` dependency

### PR DESCRIPTION
**Pull request by castroinsua**
There are two important changes in this pull request, which are also summarized in the `NEWS.md` file:

1. Update to PBDB API version 1.2 and change all the documentation of the package accordingly (this should close [#51](https://github.com/ropensci/paleobioDB/issues/51) and [#57](https://github.com/ropensci/paleobioDB/issues/57)).
2. The functions that work with rasters use now methods from the `terra` package instead of the `raster` package.

This pull request also fixes issue #50 with mapping functions not working properly or throwing errors. This bug was due to the API returning the latitude and longitude coordinates as strings and passing these values to functions that expected a numeric type.

The package documentation stated that by default the maximum number of returned records was [500](https://github.com/ropensci/paleobioDB/commit/ecdbb3e5a30a35834a50e6ec29388211387cec56), but no such limit seems to be enforced anywhere in the package, and I cannot find any specific value for this limit in the API documentation, so I assume that it returns all records that satisfy the request. Perhaps it would be a good idea to set a limit by default in the package and encourage users to download their data in chunks (maybe by using the [`offset`](https://paleobiodb.org/data1.2/special_doc.html) parameter) if the result is very large.

Some functions (e.g. `pbdb_subtaxa()`, `pbdb_temp_range()`) still do not work if a dataframe with compact field names (which is what the API returns by default) is provided, so it is necessary to call `pbdb_occurrences()` with the `vocab = "pbdb"` parameter to get a dataframe that is suitable for these functions. In the future, it would be important to either document this or alter these functions so that they can work with a dataframe with abbreviated field names. Personally, I would prefer the second approach.

Since the API version change might break code that was written using earlier versions of the package, a major version change in the package might be needed.

### CODE REVIEW

**Review by WDragon101 — APPROVED** [review]

**Review by WDragon101 — APPROVED** [review]

**Review by WDragon101 — APPROVED** [review]

**Review by WDragon101 — APPROVED** [review]

**Review by WDragon101 — APPROVED** [review]

**Review by WDragon101 — APPROVED** [review]

File: R/pbdb_querys.R [hunk 1]
```diff
@@ -761,32 +772,33 @@ pbdb_ref_collections <-function(...){
...
+#'       matching taxa.
 #'     \item \emph{extant}: TRUE/FALSE to select extant/extinct taxa.
 #'   }
 #' @return a dataframe with references from a list of taxa
 #'  
 #' @export 
 #' @examples \dontrun{
-#'   pbdb_ref_taxa (name="Canidae", vocab="pbdb", show=c("attr", "app", "size", "nav")) 
+#'   pbdb_ref_taxa(
+#'     name = "Canidae", vocab = "pbdb", show = c("attr", "app", "size", "nav")
```
Comment (WDragon101) [inline R/pbdb_querys.R:800]: Accroding to [PBDB Data Service 1.2 v2](https://paleobiodb.org/data1.2/) > [Taxonomic names](https://paleobiodb.org/data1.2/taxa_doc.html) > [Bibliographic references for taxa](https://paleobiodb.org/data1.2/taxa/refs_doc.html), available values for **parameter _show_** are follow**ing: counts, formatted, both, comments, ent, entname, crmod**.

**Review by WDragon101 — CHANGES_REQUESTED** [review]

File: R/pbdb_querys.R (hunk 1 above)
Comment (castroinsua) [reply R/pbdb_querys.R:800]: Thank you for pointing this out. It seems that the JSON object returned by the API can have a `"warnings"` property that the `paleobioDB` package simply ignores, so many of these examples, while they are not well formed, run without errors and return a dataframe (so I did not realize they were wrong). For now, I will simply correct the examples in the documentation, but the package should probably report these warnings.

File: man/pbdb_collection.Rd [hunk 2]
```diff
@@ -31,8 +31,6 @@ the main filtering parameters.
 }
 \examples{
 \dontrun{
-pbdb_collection (id=1003, vocab="pbdb", show="loc")
-
+  pbdb_collection(id = 1003, vocab = "pbdb", show = "loc")
```
Comment (WDragon101) [inline man/pbdb_collection.Rd:34]: Should be consistent with **R/pbdb_querys.R** **line 688**: pbdb_reference(id = 1003, vocab = "pbdb", show = "both")

**Review by WDragon101 — APPROVED** [review]

**Comment by WDragon101** [comment]
Should you uncomment the code on lines 64 to 70 of tests/testthat/test-all_basic_queries.R ?

**Comment by castroinsua** [comment]
> Should you uncomment the code on lines 64 to 70 of tests/testthat/test-all_basic_queries.R ?

I do not know for what particular reason these tests were commented out. However, I think that many of these tests will be changed in the future so that an internet connection is not needed to run them, so for now I will not touch this file.

**Comment by castroinsua** [comment]
@WDragon101 I think the specific issues you mentioned are now solved, but I will take more time to ensure that the package documentation is consistent with the API documentation before releasing a new version.

**Review by WDragon101 — APPROVED** [review]

**Pull request merged**